### PR TITLE
Source review and advanced section

### PR DESCRIPTION
**Pull request by dsvanlani**
## What does this PR accomplish?

- Introduces an "Advanced" section in the resource configuration form, which was previously supported on the back end but not respected.
    - Parameters are grouped with adjacent parameters that have the same value of `advancedConfig`.  This allows multiple advanced sections throughout the form.
image

- Individual source feedback, see [this doc](https://docs.google.com/document/u/1/d/1JkJIdUCN4wxwm3IQ-6QKzZgYRatJ5Oto2ABJSFRvheg/edit).
- All `Filtering` checkboxes are removed in favor of showing `metrics` parameters under the advanced section.
   - Removed validating values for in line sources against the source type.  This is needed to make sure these changes are backwards compatible with previously created configurations.
image

### CODE REVIEW

File: resources/source-types/kafka_cluster.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: SourceType
 metadata:
   name: kafka_cluster
   displayName: Kafka Cluster
-  description: Collect cluster metrics from a Kafka cluster brokers.
+  description: Collect metrics from Kafka cluster brokers
```
Comment (antonblock) [inline resources/source-types/kafka_cluster.yaml:6]:
```suggestion
  description: Collect metrics from Kafka cluster brokers.
```

File: resources/source-types/sap_hana.yaml [hunk 2]
```diff
@@ -2,10 +2,10 @@ apiVersion: bindplane.observiq.com/v1
 kind: SourceType
 metadata:
   name: hana
-  displayName: SAP Hana
+  displayName: SAP HANA
   icon: /icons/sources/sap.svg
   #description: Collect metrics logs from SAP Hana
-  description: Collect logs from SAP Hana
+  description: Collect logs from SAP HANA
```
Comment (antonblock) [inline resources/source-types/sap_hana.yaml:8]:
```suggestion
  description: Collect logs from SAP HANA.
```

File: resources/source-types/cisco_asa.yaml [hunk 3]
```diff
@@ -4,27 +4,26 @@ metadata:
   name: ciscoasa
   displayName: Cisco ASA
   icon: /icons/sources/cisco.svg
-  description: Collect logs from Cisco ASA.
+  description: Receive Syslog from Cisco ASA.
```
Comment (antonblock) [inline resources/source-types/cisco_asa.yaml:7]:
```suggestion
  description: Receive syslog from Cisco ASA.
```

File: resources/source-types/cisco_catalyst.yaml [hunk 4]
```diff
@@ -4,27 +4,27 @@ metadata:
   name: ciscocatalyst
   displayName: Cisco Catalyst
   icon: /icons/sources/cisco.svg
-  description: Collect logs from Cisco Catalyst.
+  description: Receive Syslog from Cisco Catalyst.
```
Comment (antonblock) [inline resources/source-types/cisco_catalyst.yaml:7]:
```suggestion
  description: Receive syslog from Cisco Catalyst.
```

File: resources/source-types/cisco_meraki.yaml [hunk 5]
```diff
@@ -4,27 +4,27 @@ metadata:
   name: ciscomeraki
   displayName: Cisco Meraki
   icon: /icons/sources/cisco_meraki.svg
-  description: Collect logs from Cisco Meraki.
+  description: Receive Syslog from Cisco Meraki.
```
Comment (antonblock) [inline resources/source-types/cisco_meraki.yaml:7]:
```suggestion
  description: Receive syslog from Cisco Meraki.
```

**Review by antonblock — COMMENTED** [review]
This is awesome, huge improvements. I just saw a couple tiny inconsistencies with punctuation

**Review by andykellr — APPROVED** [review]

**Review by antonblock — APPROVED** [review]

**Pull request merged**